src/assets/scripts/aircraft/AircraftFlightManagementSystem.js

```diff
@@ -286,53 +312,59 @@ export default class AircraftFlightManagementSystem {
      * Modify the current waypoint
      */
     setCurrent(data) {
+        // TODO: refactor this, what is actually happening here?
         for (const i in data) {
-            this.currentWaypoint()[i] = data[i];
+            this.currentWaypoint[i] = data[i];
         }
     }
 
     /**
      * Updates fms.fp.route to correspond with the fms Legs
      */
     update_fp_route() {
-        const r = [];
+        const flightPlanRoute = [];
 
         // TODO: simplify this
         // FIXME: is this.legs an array?
-        for (const l in this.legs) {
-            if (!this.legs[l].type) {
+        for (let i = 0; i < this.legs.length; i++) {
+            const leg = this.legs[i];
+
+            if (!leg.type) {
                 continue;
             }
 
-            switch (this.legs[l].type) {
+            // FIXME: replace the string splitting with the `RouteModel` class methods
+            switch (leg.type) {
                 case FP_LEG_TYPE.SID:
-                    // TODO: this split logic and string building should live in a helper function or or class method
                     // departure airport
-                    r.push(this.legs[l].route.split('.')[0]);
+                    flightPlanRoute.push(leg.route.entry);
                     // 'sidname.exitPoint'
-                    r.push(this.legs[l].route.split('.')[1] + '.' + this.legs[l].route.split('.')[2]);
+                    flightPlanRoute.push(`${leg.route.procedure}.${leg.route.exit}`);
 
                     break;
                 case FP_LEG_TYPE.STAR:
                     // 'entryPoint.starname.exitPoint'
-                    r.push(this.legs[l].route.split('.')[0] + '.' + this.legs[l].route.split('.')[1]);
+                    flightPlanRoute.push(`${leg.route.entry}.${leg.route.procedure}`);
                     // arrival airport
-                    r.push(this.legs[l].route.split('.')[2]);
+                    flightPlanRoute.push(leg.route.exit);
 
                     break;
                 case FP_LEG_TYPE.IAP:
                     // no need to include these in flightplan (because wouldn't happen in real life)
                     break;
                 case FP_LEG_TYPE.AWY:
```

Comment:
Yes and no; they just have to be added to the airport files. They should work fine once airports have airways added in.